src/FSharp.Data.GraphQL.Server/Values.fs

```diff
@@ -122,27 +131,41 @@ let rec internal compileByType
+                            if ReflectionHelper.isAssignableWithUnwrap inputDef.Type param.ParameterType then
+                                all.Add (struct (ValueSome field, param)) |> ignore
+                            else
+                                // TODO: Consider improving by specifying type mismatches
+                                mismatch.Add param.Name |> ignore
                     | None ->
                         if ReflectionHelper.isParameterOptional param then
                             all.Add <| struct (ValueNone, param) |> ignore
                         else
                             missing.Add param.Name |> ignore
-                    struct (all, areNullable, missing))
-                struct (ResizeArray (), HashSet (), HashSet ())
-
-        if missingParameters.Any () then
-            raise
-            <| InvalidInputTypeException (
-                $"Input object '%s{objDef.Name}' refers to type '%O{objtype}', but mandatory constructor parameters '%A{missingParameters}' don't match any of the defined input fields",
-                missingParameters.ToImmutableHashSet ()
-            )
-        if nullableMismatchParameters.Any () then
-            raise
-            <| InvalidInputTypeException (
-                $"Input object %s{objDef.Name} refers to type '%O{objtype}', but optional fields '%A{missingParameters}' are not optional parameters of the constructor",
-                nullableMismatchParameters.ToImmutableHashSet ()
-            )
+                    struct (all, mismatch, areNullable, missing))
+                struct (ResizeArray (), HashSet (), HashSet (), HashSet ())
+
+        let exceptions : exn list =  [
+            if missingParameters.Any () then
+                InvalidInputTypeException (
+                    $"Input object '%s{objDef.Name}' refers to type '%O{objtype}', but mandatory constructor parameters '%A{missingParameters}' don't match any of the defined input fields",
+                    missingParameters.ToImmutableHashSet ()
+                )
+            if nullableMismatchParameters.Any () then
+                InvalidInputTypeException (
+                    $"Input object %s{objDef.Name} refers to type '%O{objtype}', but optional fields '%A{missingParameters}' are not optional parameters of the constructor",
+                    nullableMismatchParameters.ToImmutableHashSet ()
+                )
+            if typeMismatchParameters.Any () then
+                InvalidInputTypeException (
+                    $"Input object %s{objDef.Name} refers to type '%O{objtype}', but fields '%A{typeMismatchParameters}' have different types than constructor parameters",
+                    typeMismatchParameters.ToImmutableHashSet ()
+                )
+        ]
+        match exceptions with
+        | [] -> ()
+        | [ ex ] -> raise ex
+        | _ -> raise (AggregateException ($"Invalid input object '%O{objtype}'", exceptions))
```

Comment:
And throw aggregate exception if multiple issues appear in an input object definition